cartography/intel/dns.py

```diff
@@ -79,23 +81,24 @@ def ingest_dns_record(neo4j_session, name, value, type, update_tag, points_to_re
     :param type: record type
     :param update_tag: Update tag to set the node with and childs
     :param points_to_record: parent record to set DNS_POINTS_TO relationship to. Can be None
+    :param node_label: the label of the node to attach to a DNS record
     :return: the intel graph node id for the new/merged record
     """
-    ingest = """
+    template = Template("""
     MERGE (record:DNSRecord{id: {Id}})
     ON CREATE SET record.firstseen = timestamp(), record.name = {Name}, record.type = {Type}
     SET record.lastupdated = {update_tag}, record.value = {Value}
     WITH record
-    MATCH (n{id: {PointsToId}})
+    MATCH (n:$node_label{id: {PointsToId}})
```

Comment:
can you add notes on why the node_label is needed since the node id is specified and expected to be unique?

Reply:
ok so perf reasons. Just curiousity, would love to see perf stats on that if you have on if the indexing is applied with or without labeling.

Reply:
@sachafaust I just learned this the other day, it's pretty interesting!  If you `match` on an indexed ID, Neo4j performs a NodeIndexSeek and processes just 1 estimated row so performance should be fast:
![image](https://user-images.githubusercontent.com/46503781/78855588-2bec6c00-79d9-11ea-8f36-e02ef318eb26.png)

However, without node labels Neo4j has to do an AllNodesScan, which in my case here involves over 2 million rows:
![image](https://user-images.githubusercontent.com/46503781/78855648-55a59300-79d9-11ea-8e77-2c0f69b825f7.png)

As Neo4j docs say, we should [avoid AllNodesScan as much as possible](https://neo4j.com/docs/cypher-manual/current/execution-plans/operators/#query-plan-all-nodes-scan).